### ISSUE TITLE
Magic variables warning

### ISSUE THREAD

**Issue by rebelvg** [LordGolias/sqf#16]
```
private _code = {side _x == WEST};

allPlayers select _code
```

This will produce a warning about variable scope for _x on the first line, this is kinda correct, no way to known if _x will be set as magic variable at this stage.
Warning can be fixed by adding private ["_x"].

Not sure if that can be called an issue, just more of a question if that's right or wrong.

**Comment by LordGolias** [comment]
Yes, this is a valid issue and a valid use case, thanks for bringing it up!

Currently, assigned code (i.e. code that is assigned to a variable) is analyzed with `_this` declared only.

Another possibility is to analyze assigned code with common variables declared (e.g. `_x`, `_thisScript`), instead of just `_this`. This however can cause the analyzer to miss cases where `_x` is used without private (which is dangerous because `_x` is used in numerous contexts).

I would say that both possibilities have their advantages and disadvanges. Either miss non-private `_x`, or incorrectly warn about `_x`.

What do you think?

**Comment by rebelvg** [comment]
I see arguments for both sides. Not everywhere _x is a magic variable so warning is correct, on the other hand it's common practice that _x is reserved. Really not sure about this one.

**Comment by jonpas** [comment]
Can you analyse code in variable only where it is used?

**Comment by SilentSpike** [comment]
Also just ran into this issue and was about to report it :+1: Had the same question as jonpas above

**Comment by LordGolias** [comment]
@jonpas it is possible, yes, just not implemented at the moment.

Currently, the analyzer only tracks variable types (e.g. `Number`, `String`, `Code`). Already this is a challenge (issue #15). Once that is working, we can start tracking variables `values` (e.g. `Number(1)`).

Once we track values, we can implement what you are suggesting, because `_code` in `select _code` will contain the assigned code, instead of just the type.

It does not solve the whole issue as branching will cause problems, e.g. 

```
if (_z) then {
    _code = {_x == 2};
} else {
    _code = {_x == 3};
};
_z1 select _code;
```
